apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -341,28 +371,72 @@ impl ApplyTo for PathSelection {
     }
 }
 
+impl ApplyTo for LitExpr {
+    fn apply_to_path(
+        &self,
+        data: &JSON,
+        vars: &VarsWithPathsMap,
+        input_path: &InputPath<JSON>,
+        errors: &mut IndexSet<ApplyToError>,
+    ) -> Option<JSON> {
+        match self {
+            Self::String(s) => Some(JSON::String(s.clone().into())),
+            Self::Number(n) => Some(JSON::Number(n.clone())),
+            Self::Bool(b) => Some(JSON::Bool(*b)),
+            Self::Null => Some(JSON::Null),
+            Self::Object(map) => {
+                let mut output = JSONMap::with_capacity(map.len());
+                for (key, value) in map {
+                    if let Some(value_json) = value.apply_to_path(data, vars, input_path, errors) {
+                        output.insert(key.clone(), value_json);
+                    }
```

Comment:
What happens if `value.apply_to_path(...)` errors here? Will the original map just have the values which errored not get inserted into the resulting map at all? If we're copying javascript here, that would be the equivalent to setting the key's value to `undefined`, not null right? Will the execution fill in the missing key here?

Reply:
If/when `value.apply_to_path` returns `None`, then the property will not be inserted into `output`, but it's expected that an error should have been added to the `errors` set to explain the absence of the property (assuming an error is warranted).

In practice, if you queried for this field with GraphQL, it would become `null` in the final result (instead of missing), along with an error in `errors`. Which is what you'd expect from GraphQL-style error handling, I think.